src/H5Dchunk.c

```diff
@@ -8066,13 +8067,20 @@ static int
 H5D__chunk_iter_cb(const H5D_chunk_rec_t *chunk_rec, void *udata)
 {
     const H5D_chunk_iter_ud_t *data      = (H5D_chunk_iter_ud_t *)udata;
+    const H5O_layout_chunk_t  *chunk     = data->chunk;
     int                        ret_value = H5_ITER_CONT;
+    hsize_t                    offset[H5O_LAYOUT_NDIMS];
+    unsigned                   ii; /* Match H5O_layout_chunk_t.ndims */
+
+    /* Similar to H5D__get_chunk_info */
+    for (ii = 0; ii < chunk->ndims; ii++)
```

Comment:
I think this is off by one?  The `ndims` here is 1 larger than the rank I set. I also see in other places, `layout->ndims` is decreased by 1 to determine the number of dimensions.
https://github.com/HDFGroup/hdf5/blob/509fe962d42b3f02d356d2668b04712df06daf7f/src/H5Dearray.c#L1095-L1100

Reply:
If you add an assertion here and run [this test](https://github.com/HDFGroup/hdf5/blob/509fe962d42b3f02d356d2668b04712df06daf7f/test/chunk_info.c#L1579), it's easy to see that `chunk->ndims` is 3 but we clearly expect the `offset` array to have two valid elements.

Reply:
You are correct that there are ndims - 1 dimensions, but this is an internal implementation detail of the library. That specific loop comes from !`H5D__get_chunk_info`.

What is the actual problem though? At worse, we do one more multiplication than needed while avoiding a subtraction.